event/handler.go

```diff
@@ -204,8 +208,18 @@ func (handler *ExportHandler) fullDownload(event *FilterSubmitted, isPublished b
 		return nil, err
 	}
 
-	downloads := map[string]dataset.Download{
-		"CSV": {Size: strconv.Itoa(int(reader.TotalBytesRead())), URL: fileURL},
+	downloads := make(map[string]dataset.Download)
+	downloadURL := fmt.Sprintf("%s/downloads/datasets/%s/editions/%s/versions/%s.csv",
+		handler.downloadServiceURL,
+		event.DatasetID,
+		event.Edition,
+		event.Version,
+	)
+
+	if isPublished {
+		downloads["CSV"] = dataset.Download{Size: strconv.Itoa(int(reader.TotalBytesRead())), Public: fileURL, URL: downloadURL}
```

Comment:
might be worth logging which case is being applied here - if we ever need to be 100% sure whats happening

Reply:
Log line added below 👍